wmi_check/datadog_checks/wmi_check/wmi_check.py

```diff
@@ -14,68 +15,49 @@ class WMICheck(WinWMICheck):
-        namespace = instance.get('namespace', "root\\cimv2")
-        provider = instance.get('provider')
-        username = instance.get('username', "")
-        password = instance.get('password', "")
-
-        # WMI instance
-        wmi_class = instance.get('class')
-        metrics = instance.get('metrics')
-        filters = instance.get('filters')
-        tag_by = instance.get('tag_by', "")
-        tag_queries = instance.get('tag_queries', [])
-
-        constant_tags = instance.get('constant_tags')
-        custom_tags = instance.get('tags', [])
-        if constant_tags is None:
-            constant_tags = list(custom_tags)
-        else:
-            constant_tags.extend(custom_tags)
+        constant_tags = self.instance.get('constant_tags', [])
+        if constant_tags:
             self.log.warning("`constant_tags` is being deprecated, please use `tags`")
+        constant_tags.extend(self.custom_tags)
 
         # Create or retrieve an existing WMISampler
-        instance_hash = hash_mutable(instance)
-        instance_key = self._get_instance_key(host, namespace, wmi_class, instance_hash)
-
-        metric_name_and_type_by_property, properties = self._get_wmi_properties(instance_key, metrics, tag_queries)
+        metric_name_and_type_by_property, properties = self.get_wmi_properties()
 
-        wmi_sampler = self._get_running_wmi_sampler(
-            instance_key,
-            wmi_class,
-            properties,
-            tag_by=tag_by,
-            filters=filters,
-            host=host,
-            namespace=namespace,
-            provider=provider,
-            username=username,
-            password=password,
-        )
+        wmi_sampler = self.get_running_wmi_sampler(properties, self.filters)
 
         # Sample, extract & submit metrics
         try:
             wmi_sampler.sample()
-            metrics = self._extract_metrics(wmi_sampler, tag_by, tag_queries, constant_tags)
+            extracted_metrics = self.extract_metrics(constant_tags, wmi_sampler)
         except TimeoutException:
             self.log.warning(
                 "WMI query timed out. class=%s - properties=%s - filters=%s - tag_queries=%s",
-                wmi_class,
+                self.wmi_class,
                 properties,
-                filters,
-                tag_queries,
+                self.filters,
+                self.tag_queries,
             )
         else:
-            self._submit_metrics(metrics, metric_name_and_type_by_property)
+            self._submit_metrics(extracted_metrics, metric_name_and_type_by_property)
+
+    def extract_metrics(self, constant_tags, wmi_sampler):
+        # type: (List[str], WMISampler) -> List[WMIMetric]
+        return self._extract_metrics(wmi_sampler, self.tag_by, self.tag_queries, constant_tags)
+
+    def get_wmi_properties(self):
+        # type: () -> WMIProperties
+        return self._get_wmi_properties(None, self.metrics_to_capture, self.tag_queries)
```

Comment:
`instance_key` seems not used anymore, maybe we should get rid of it as base level: https://github.com/DataDog/integrations-core/blob/1b52f8161ad9a96ba6222ff46a9889a3971fa66a/datadog_checks_base/datadog_checks/base/checks/win/wmi/__init__.py#L270-L297

Reply:
Yes, but I cannot remove from base class until is removed from all subclasses

Reply:
and its's hard cause the way to remove it would be to have default options for all arguments (since its the first one):
```
def _get_running_wmi_sampler(self, instance_key=None, wmi_class=None, properties=None, tag_by="", **kwargs=None)
```
then subclasses should use keyword arguments, then I could remove it from base class.
I don't think it's worth the hassle. 
```

Reply:
Yes, you are right, it's better to removed it later is another PR 👍